### PR TITLE
many: fix shellcheck warnings in bionic

### PR DESCRIPTION
**Pull request by mvo5**
On my bionic system I saw a bunch of shellcheck errors. This PR fixes those.

### CODE REVIEW

File: cmd/snap-mgmt/snap-mgmt.sh.in [hunk 1]
```diff
@@ -33,7 +33,7 @@ systemctl_stop() {
 }
 
 purge() {
-    # debian-9, fedora-27, ubuntu-16.04...
+    # shellcheck disable=SC1091
     distribution=$(. /etc/os-release; echo "${ID}-${VERSION_ID}")
```
Comment (zyga) [inline cmd/snap-mgmt/snap-mgmt.sh.in]: Can we just do this:
distribution="$(. /etc/os-release; echo "${ID}-${VERSION_ID}")"

File: tests/lib/pkgdb.sh [hunk 2]
```diff
@@ -205,7 +205,7 @@ distro_install_package() {
 }
 
 distro_purge_package() {
-    set -- $(
+    set -- "$(
```
Comment (zyga) [inline tests/lib/pkgdb.sh]: I _think_ this is broken now

File: cmd/snap-mgmt/snap-mgmt.sh.in (hunk 1 above)
Comment (bboozzoo) [reply cmd/snap-mgmt/snap-mgmt.sh.in]:
```
distribution="$(. /etc/os-release; echo ${ID}-${VERSION_ID})"
```
should be fine for shellcheck

Comment (mvo5) [reply cmd/snap-mgmt/snap-mgmt.sh.in]: It is unhappy about following a source that is outside the source tree, the quoting is ok

**Pull request merged**